src/data/datasets/representation/db_retrieval_dataset.py

```diff
@@ -0,0 +1,152 @@
+                 db_folder: str,
+                 include_only_path: str,
+                 img_paths_csv_path: str,
+                 transform: Union[BasicTransform, BaseCompose],
+                 augment: Optional[Union[BasicTransform, BaseCompose]] = None,
+                 input_dtype: str = 'float32',
+                 input_column: str = 'image_path',
+                 grayscale: bool = False):
+        """
+        Args:
+            data_folder: Directory with all the images.
+            matches_csv_path: path to csv file where queries with their relevance scores are specified
+            db_folder: path to a folder with all database images (traversed recursively)
+            include_only_path: sometext
+            img_paths_csv_path: path to mapping image identifiers to image paths. Format: id | path.
+            transform: Transform to be applied on a sample. This should have the
+                interface of transforms in `albumentations` library.
+            augment: Optional augment to be applied on a sample.
+                This should have the interface of transforms in `albumentations` library.
+            input_dtype: Data type of of the torch tensors related to the image.
+            input_column: Name of the column that contains paths to images.
+            grayscale: If True, image will be read as grayscale otherwise as RGB.
+        """
+        super().__init__(data_folder, matches_csv_path, img_paths_csv_path,
+                         transform, augment, input_dtype, input_column, grayscale)
+        self.__db_folder = Path(db_folder)
+
+        self.__include_only = pd.read_csv(Path(db_folder) / include_only_path, usecols=["id", input_column],
+                                          dtype={"id": str, input_column: str}, header=0)
+
+        self.__include_only[input_column] = self.__include_only[input_column].map(lambda path: self.__db_folder / path)
+        self.__db_objid2paths = self.__include_only.groupby('id')[input_column].apply(list).to_dict()
+        self._objid2paths.update(self.__db_objid2paths)
+
+        self.__n_db = 0
+        for obj_id in self.__db_objid2paths.keys():
+            self._index2objid[self._data_len + self.__n_db] = obj_id
+            self.__n_db += 1
+
+        self._data_len += self.__n_db
+        self.__objid2index = {}
+
+        for index, obj_id in self._index2objid.items():
+            if obj_id not in self.__objid2index:
+                self.__objid2index[obj_id] = index
+
+        self.__relevance_map = self.__get_relevance_scores()
+        self.__targets = np.zeros((len(self), 1 + self._n_queries), dtype=np.float32)
+
+        for index in range(self._n_queries):
+            relevant_idxs = self.__relevance_map[index]['relevant_idxs']
+            relevant_scores = self.__relevance_map[index]['relevant_scores']
+            self.__targets[relevant_idxs, 1 + index] = relevant_scores
+            self.__targets[index, 0] = 1.
```

Comment:
Make two separate tensors at output: one for relevant scores (`scores`), another is a boolean tensor of query indicators (`is_query`)